### PR TITLE
Homepage showcase block

### PR DESCRIPTION
**Pull request by helenb**
[Link to Ticket](https://torchbox.monday.com/boards/1192293412/pulses/1435123100)

### Description of Changes Made

- Adds a new block called 'homepage showcase block' - just for the home page
- Shares a template and CSS with the original showcase block with a few variations
- Icons are static for now - there is a post-launch ticket to animate them
- Ensure that the original showcase block is not affected by the changes, apart from a tweak to the gap between the ring icon and text at mobile.

### How to Test

- Add a 'homepage showcase block' on the home page and view that, or in the styleguide of the pattern library http://localhost:8000/pattern-library/render-pattern/patterns/styleguide/components/components.html

### Screenshots

<details>
  <summary>Expand to see more</summary>

Desktop

Screenshot 2024-03-26 at 14 32 39

Mobile

Screenshot 2024-03-26 at 14 33 13

</details>

### MR Checklist

- [x] Add a description of your pull request and instructions for the reviewer to verify your work.
- [x] If your pull request is for a specific ticket, link to it in the description.
- [x] Stay on point and keep it small so the merge request can be easily reviewed.
- [x] Tests and linting passes.

#### Unit tests

- [ ] Added
- [x] Not required

#### Documentation

- [ ] Updated build docs
- [ ] Updated editor guidelines (https://docs.google.com/document/d/1PAWccdQ4tfaZsrEWmpDhvP3GH5RRmBOARFVp4b-kje8/edit?usp=sharing)
- [x] Not required
Note: I will update the fieldspec.

#### Browser testing

- [x] I have tested in the following browsers and environments (edit the list as required)
  - Latest version of Chrome on mac
  - Latest version of Firefox on mac
  - Latest version of Safari on mac
  - Safari on last two versions of iOS
  - Chrome on last two versions of Android
- [ ] Not required

#### Data protection

- [x] Not relevant
- [ ] This adds new sources of PII and documents it and modifies Birdbath processors accordingly

#### Accessibility

- [x] Automated WCAG 2.1 tests pass
- [x] HTML validation passes
- [x] Manual WCAG 2.1 tests completed
- [x] I have tested in a screen reader
- [x] I have tested in high-contrast mode
- [x] Any animations removed for prefers-reduced-motion
- [ ] Not required

#### Sustainability

- [ ] Images are optimised and lazy-loading used where appropriate
- [x] SVGs have been optimised
- [ ] Perfomance and transfer of data considered
- [ ] If JavaScript is needed alternatives have been considered
- [ ] Not required

#### Pattern library

- [x] The pattern library component for this template displays correctly, and does not break parent templates (Note that the pattern library shows the original showcase block)
- [x] The styleguide is updated if relevant
- [ ] Changes are not relevant the pattern library

### CODE REVIEW

File: tbx/project_styleguide/templates/patterns/molecules/streamfield/blocks/showcase_block.md [hunk 1]
```diff
@@ -4,3 +4,4 @@
 - Each item shows a title and text and can have an optional link.
 - The example here shows items with an without links for demonstration purposes but in reality you would either have all items with links or all items without links.
 - The icons used to the left of the item should update with theme choice.
+- There is a slightly different version of the showcase block that is used on the home page. It is a different block but shares the template and styling due to similar design.
```
Comment (engineervix) [inline tbx/project_styleguide/templates/patterns/molecules/streamfield/blocks/showcase_block.md:7]: **Praise**: Thanks for documenting this!

File: tbx/project_styleguide/templates/patterns/molecules/streamfield/blocks/showcase_block.html [hunk 2]
```diff
@@ -1,27 +1,50 @@
...
                     <a href="{% pageurl item.page %}" class="showcase__item showcase__item--link" aria-label="{{ item.heading }}">
                 {% endif %}
 
-                <div class="showcase__rings">
-                    {% include "patterns/atoms/icons/icon.html" with name="ring-one" classname="icon--stroke showcase__ring showcase__ring--one" %}
-                    {% include "patterns/atoms/icons/icon.html" with name="ring-two" classname="icon--stroke showcase__ring showcase__ring--two" %}
-                </div>
+                {# Showcase block uses the rings icons, homepage block uses an icon selected by the editor #}
+                {% if is_homepage_showcase %}
+                    {# Note that a future enhancement will be to animated these icons, in which case they will need to be split out into separate svgs and a custom css animation applied to each one - similar to the 'employee owned trust' icon on the home page #}
```
Comment (engineervix) [inline tbx/project_styleguide/templates/patterns/molecules/streamfield/blocks/showcase_block.html:24]: :+1:

**Review by engineervix — COMMENTED** [review]
Thanks Helen, this is looking great! Just a few BE suggestions for consideration

**Review by chris-lawton — APPROVED** [review]

File: tbx/core/blocks.py [hunk 3]
```diff
@@ -248,10 +249,58 @@ class ShowcaseBlock(blocks.StructBlock):
 
     class Meta:
         icon = "tasks"
+        # Same template as the homepage showcase block
         template = "patterns/molecules/streamfield/blocks/showcase_block.html"
         group = "Custom"
 
 
+class IconChoice(models.TextChoices):
```
[the remark this answers is not in the document]
Comment (helenb) [reply tbx/core/blocks.py:257]: I don't think so at the moment - it's very specific to that block. We can refactor later if needed.

File: tbx/core/blocks.py [hunk 4]
```diff
@@ -649,6 +698,7 @@ class Meta:
 
 class HomePageStoryBlock(blocks.StreamBlock):
     showcase = ShowcaseBlock()
+    homepage_showcase = HomepageShowcaseBlock()
```
[the remark this answers is not in the document]
Comment (helenb) [reply tbx/core/blocks.py:701]: sounds sensible!

File: tbx/core/blocks.py [hunk 5]
```diff
@@ -248,10 +249,58 @@ class ShowcaseBlock(blocks.StructBlock):
...
 
 
+class IconChoice(models.TextChoices):
+    LIGHTBULB = "lightbulb", "lightbulb icon"
+    TARGET = "target", "target icon"
+    MEGAPHONE = "megaphone", "megaphone icon"
+    WAGTAIL = "wagtail", "wagtail icon"
+
+
+class HomepageShowcaseBlock(blocks.StructBlock):
```
[the remark this answers is not in the document]
Comment (helenb) [reply tbx/core/blocks.py:269]: done!

**Pull request merged**